docs/reference/deployment-targets-yaml-reference.md

```diff
@@ -0,0 +1,271 @@
+the deployment target. It should not be modified.
+
+#### created_by
+
+The `created_by` property contains the UUID of the user who created the deployment
+target. It should not be modified.
+
+#### created_at
+
+The `created_at` property indicates the UTC when the deployment target was created.
+It should not be modified.
+
+#### updated_by
+
+The `updated_by` property stores the UUID of the user who last updated the deployment
+target. It should not be modified.
+
+#### updated_at
+
+The `updated_at` property denotes the UTC when the deployment target was last
+updated. It should not be modified.
+
+#### description
+
+The `description` property is a string that describes the deployment target.
+
+#### url
+
+The `url` property is a string containing the URL associated with the deployment target.
+
+### spec
+
+The `spec` property contains a list of deployment target properties.
+
+#### state
+
+The `state` property indicates the state of the deployment target, which can be
+one of the following values: `SYNCING`, `USABLE`, `UNUSABLE`, or `CORDONED`. It should
+not be modified.
+
+#### state_message
+
+The `state_message` property provides an additional message associated with
+the deployment target's state. It should not be modified.
+
+#### subject_rules
+
+The `subject_rules` property holds a list of subject rules that determine who
+can trigger promotions when the deployment target is associated with them. Each
+rule must include the `type`. Depending on the rule `type`, the `subject_id`
+or `git_login` may also be required.
+
+##### type
+
+The `type` property of a subject rule can have the following values:
+`ANY`, `USER`, `ROLE`, `GROUP`, or `AUTO`. If `ANY` is used, any user can trigger
+the promotion. If `USER` is used, only specified users (who must be assigned to
+the project) can trigger a promotion. For `ROLE`, only specified user roles defined
+within the project can trigger a promotion.
+
+##### subject_id
+
+The `subject_id` property is required for `USER`, `ROLE`, and `GROUP` rule types.
+For `USER`, the `subject_id` contains the UUID of the user. If the rule type is `ROLE`,
+it should be the name of the project role (e.g., `Admin`, `Contributor`).
+
+##### git_login
+
+The `git_login` property is used only for `USER` rule types when specifying a user
+by their git handle (e.g., GitHub login) instead of their UUID.
+
+#### object_rules
+
+The `object_rules` property contains a list of object rules that define which branches,
+tags, or pull requests can trigger promotions. Each rule must include the `type`,
+`match_mode`, and `pattern` properties.
+
+##### type
+
+The `type` property of an object rule can have the following values: `BRANCH`, `TAG`,
+or `PR`.
+
+##### match_mode
+
+The `match_mode` property determines how the pattern matches the name of the git
+reference (branch, tag, or pull request). The `match_mode` can be one of the following:
+`ALL`, `EXACT`, or `REGEX`.
```

Comment:
For pull requests currently there is only "on/off" support and there is no way to do exact or regex matches on anything there.
Please check how is that translated in this case and add that information here.